### PR TITLE
QUIC: Assorted Tests 1, Connection Closure Reason

### PR DESCRIPTION
**Pull request by hlandau**
Built on miscellaneous conformance updates PR.

This also finally wires up the connection closure reason string in the API.

New commits in this PR:

```
eb52c1fa33 QUIC TEST: Connection closure reason testing
4f04df9e8f QUIC APL/CHANNEL: Wire up connection closure reason
d70e7982b7 QUIC TEST: Test repeated HANDSHAKE_DONE, non-minimal encoding
d6d619f475 QUIC TEST: Test NEW_CONN_ID frames
b503c93c80 QUIC TEST: Test malformed crypto stream data, excess buffering
a5b6c6be70 QUIC TEST: Ensure bogus BLOCKED frames are ignored
422bcecf36 QUIC TEST: Ensure PATH_RESPONSE is ignored
126e28a7f8 QUIC TEST: Test ACK frame generation
9f6aa9c751 QUIC TEST: Ensure PING causes ACK generation
e0113e80f3 QUIC TEST: Fault Injection: PADDING
```

Fixes https://github.com/openssl/project/issues/75
Fixes https://github.com/openssl/project/issues/77
Fixes https://github.com/openssl/project/issues/81
Fixes https://github.com/openssl/project/issues/82
Fixes https://github.com/openssl/project/issues/86
Fixes https://github.com/openssl/project/issues/85

Fixes https://github.com/openssl/project/issues/80

Fixes https://github.com/openssl/project/issues/133

More tests to come.

### CODE REVIEW

**Comment by hlandau** [comment]
```
2ca869eb88 fixup! QUIC TEST: RESET_STREAM, STOP_SENDING
16de4a7b01 QUIC TEST: RESET_STREAM, STOP_SENDING
e20794c546 fixup! QUIC TEST: Connection closure reason testing
```

Fixes https://github.com/openssl/project/issues/80

**Comment by hlandau** [comment]
Added coverity fixes (see https://github.com/openssl/project/issues/108)

**Comment by h-vetinari** [comment]
> Fixes https://github.com/openssl/project/issues/75
> Fixes https://github.com/openssl/project/issues/77
> Fixes https://github.com/openssl/project/issues/81
> Fixes https://github.com/openssl/project/issues/82
> Fixes https://github.com/openssl/project/issues/86
> Fixes https://github.com/openssl/project/issues/85
> Fixes https://github.com/openssl/project/issues/80

Is there a compelling reason for not keeping the issue tracking public?

**Comment by hlandau** [comment]
> > Fixes [openssl/project#75](https://github.com/openssl/project/issues/75)
> > Fixes [openssl/project#77](https://github.com/openssl/project/issues/77)
> > Fixes [openssl/project#81](https://github.com/openssl/project/issues/81)
> > Fixes [openssl/project#82](https://github.com/openssl/project/issues/82)
> > Fixes [openssl/project#86](https://github.com/openssl/project/issues/86)
> > Fixes [openssl/project#85](https://github.com/openssl/project/issues/85)
> > Fixes [openssl/project#80](https://github.com/openssl/project/issues/80)
> 
> Is there a compelling reason for not keeping the issue tracking public?

We only just created this issue tracking repository. We intend to make it public soon. It is absolutely the intention that this become public.

**Comment by hlandau** [comment]
CI does not appear relevant.

**Comment by mattcaswell** [comment]
Can this be rebased?

**Comment by hlandau** [comment]
Done.

**Comment by mattcaswell** [comment]
Seems to be lots of failing CIs.

**Review by mattcaswell — CHANGES_REQUESTED** [review]

**Comment by mattcaswell** [comment]
Needs a rebase

**Review by t8m — CHANGES_REQUESTED** [review]

**Review by mattcaswell — APPROVED** [review]
I'm ok with this subject to @t8m's comments being addressed/responded to

File: ssl/quic/quic_channel.c [hunk 1]
```diff
@@ -2622,19 +2626,50 @@ int ossl_quic_channel_on_handshake_confirmed(QUIC_CHANNEL *ch)
...
+
+    dst->reason     = NULL;
+    dst->reason_len = 0;
+
+    if (src->reason != NULL && src->reason_len > 0) {
+        size_t l = src->reason_len;
+        char *r;
+
+        if (l >= SIZE_MAX)
+            --l;
```
[the remark this answers is not in the document]
Comment (hlandau) [reply ssl/quic/quic_channel.c:2645]: `l + 1` on the subsequent line could overflow.

An application could pass a `SIZE_MAX` reason string. Absurd perhaps, but I think it's strange to make this code less correct.

**Comment by hlandau** [comment]
Updated

**Review by t8m — APPROVED** [review]

**Review by mattcaswell — APPROVED** [review]

**Comment by mattcaswell** [comment]
Just reconfirming my approval following the trivial fix. I consider the 24 hour timer to have started from  @t8m's approval.

**Comment by hlandau** [comment]
Merging now.

**Comment by hlandau** [comment]
Merged to master. Thanks for the reviews.

**Pull request closed without merging**